tests/integration/spreedcheats/lib/AppInfo/Application.php

```diff
@@ -21,4 +21,23 @@
  *
  */
 
-new \OCP\AppFramework\App('spreedcheats');
+namespace OCA\SpreedCheats\AppInfo;
+
+use OCP\AppFramework\App;
+use OCP\AppFramework\Bootstrap\IBootContext;
+use OCP\AppFramework\Bootstrap\IBootstrap;
+use OCP\AppFramework\Bootstrap\IRegistrationContext;
+
+class Application extends App implements IBootstrap {
+	public const APP_ID = 'spreedcheats';
+	
+	public function __construct() {
+		parent::__construct(self::APP_ID);
+	}
+
+	public function register(IRegistrationContext $context): void {
```

Comment:
strange that this is not already defined in the base class to make it optional

but maybe it's very rare to have an app that uses zero services :smile:

Reply:
well only the IBootstrap is using this.
Also we throw a more aggressive warning when you use app.php and IBootstrap.